processing/src/test/java/org/apache/druid/segment/UnnestStorageAdapterTest.java

```diff
@@ -516,6 +516,30 @@ public void test_not_pushdown_not_filter()
     );
   }
 
+  @Test
+  public void testCannotPushdown()
+  {
+    final Filter testQueryFilter = and(ImmutableList.of(
+        not(selector(OUTPUT_COLUMN_NAME, "3")),
+        or(ImmutableList.of(
+            or(ImmutableList.of(
+                selector("newcol", "2"),
+                selector(COLUMNNAME, "2"),
+                and(ImmutableList.of(
+                    selector("newcol", "3"),
+                    selector(COLUMNNAME, "7")
+                ))
+            )),
+            selector(OUTPUT_COLUMN_NAME, "1")
+        ))
+    ));
+    testComputeBaseAndPostUnnestFilters(
+        testQueryFilter,
```

Comment:
in a sense pushdown is still happening here, albeit not the entire thing. Would have been good if we can have a filter on an expression involving more than one column to show that the pushdown in basically nil. But that can be a bit complicated

Reply:
oops, i started to write a test by copying a test and then didn't change anything, my bad, will fix